### PR TITLE
Added instructions for gh-pages deployment

### PR DESCRIPTION
**Pull request by noroadsleft**
re: #90 

Adds a document detailing how to deploy an instance of the Configurator via GitHub Pages, with a link to it on the main readme.

@mechmerlin can maybe lay off @yanfali now. :laughing:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,5 +11,5 @@ The QMK Configurator allows simple keymap creation and saving via .json keymap f
 This project is very much a work in progress. To begin contributing, please refer to the following:
 
 [Issues](https://github.com/qmk/qmk_configurator/issues)  
-[Development Environment Setup](https://github.com/qmk/qmk_configurator/blob/master/assets/js/development.md)   
-
+[Development Environment Setup](https://github.com/qmk/qmk_configurator/blob/master/assets/js/development.md)  
+[Deploying an instance of the Configurator](https://github.com/qmk/qmk_configurator/blob/master/assets/js/gh-pages_setup.md)
```
Comment (yanfali) [inline README.md]: add 'to github pages' to title

File: assets/js/gh-pages_setup.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## 1. Fork the qmk/qmk_configurator repository
+
+Go to [qmk/qmk_configurator](https://github.com/qmk/qmk_configurator), and click the `fork` button at top right.
+
+## 2. Create a gh-pages branch
+
+In your git CLI, enter:
+
+    git branch gh-pages
```
Comment (yanfali) [inline assets/js/gh-pages_setup.md]: I think this should be `git checkout gh-pages`

File: assets/js/gh-pages_setup.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+In the root directory, open the `_config.yml` file in a text editor, and edit it as appropriate:
+
+    title: QMK Configurator
+    email: <github_email>     # the email attached to your GitHub account
+    show_downloads: true
+    description: QMK for potatoes - an open source configurator for QMK Firmware (only the AVR boards right now)
+    baseurl: "/qmk_configurator"     # DO NOT CHANGE THIS LINE!
+    # Replace <github_username> on the following three lines with your GitHub username
+    url: "http://<github_username>.qmk.fm"
```
Comment (yanfali) [inline assets/js/gh-pages_setup.md]: I think it's always `<github_username>.github.io`

File: assets/js/gh-pages_setup.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+## 5. Deploy your instance
+
+When you're done editing, save the file, commit it to your branch, and push your changes to your GitHub repo.
+
+    git add .
+    git commit -m "initial setup"
+    git push
+
+Once you've pushed your changes, your instance should be live in about 30 seconds at `https://<github_username>.qmk.fm/qmk_configurator`.
```
Comment (yanfali) [inline assets/js/gh-pages_setup.md]: s/.qmk.fm/github.io/ and I think you have it.

File: assets/js/gh-pages_setup.md (hunk 3 above)
Comment (noroadsleft) [reply assets/js/gh-pages_setup.md]: Hmm, I may have edited this wrongly when I did my initial deployment. I have `http://noroadsleft.qmk.fm` but I haven't found any unusual behaviour (granted, I haven't really done much with it).

**Comment by mechmerlin** [comment]
I think you should add instructions on how to test this with two different features, similar to how Yanfali was testing our secret combo and other bug fixes. He was able to switch between the two or three different features with ease.

**Comment by noroadsleft** [comment]
@mechmerlin I have no idea how he did that, TBH. I basically click buttons and pray it all works.

Unless I'm misunderstanding your suggestion.

File: assets/js/gh-pages_setup.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+In your git CLI, enter:
+
+    git checkout gh-pages
+
+This will create a copy of the `gh-pages` branch that is already on the QMK repo.
+
+## 3. Remove the CNAME directory
+
+Remove the `CNAME` directory, which is only for the production version of the Configurator.
```
Comment (yanfali) [inline assets/js/gh-pages_setup.md]: oops, sorry, I forgot to write something here. It's a file, not a directory.

**Comment by yanfali** [comment]
@mechmerlin I think that's out of scope for this document. That's basically up to the developer to merge their changes into the gh-pages branch. I usually do this via cherry picking.

**Comment by yanfali** [comment]
@noroadsleft sorry, I missed one issue. Quick fix I promise s/directory/file/ and then we're ready to merge.

**Comment by noroadsleft** [comment]
@yanfali Got it, finally. (Was wrapped up in the Ergodox fixes on _firmware.)

**Comment by yanfali** [comment]
LGTM. Thank you!

**Pull request merged**